### ISSUE TITLE
aria-hidden="true" on body makes the whole page inaccessible

### ISSUE THREAD

**Issue by Julia1996** [reactjs/react-modal#863]
### Summary:

I've found that by default the package adds aria-hidden="true" on body element, which makes the whole page inaccessible because modal dialog is placed inside of body. According to specification (https://www.w3.org/TR/wai-aria-practices-1.1/#dialog_roles_states_props) aria-hidden is not needed at all as aria-modal="true" already prevents users of assistive technologies from perceiving content outside the dialog. 
Because of this 'aria-hidden' attribute it is easy to make a modal completelly inaccessible by mistake, if a user doesn't call .setAppElement with a container that is not a body. Why the package adds this attribute, if it is not required by specification?

### Steps to reproduce:

 1. Use react-modal not calling setAppElement and not setting ariaHideApp={false}.
 2. Open modal.
 3. Observe body tag, that has 'aria-hidden=true'.

### Expected behavior:

body doens't have such an attribute.

**Comment by diasbruno** [comment]
Hi @Julia1996, yes...it's true. I would like to remove the default in this case, but I don't know if it would break someone's code.

If we are going to do something about this, it think the right way is to bump a major version to treat as a breaking change.

Also, I'd love to hear other opinions regarding this issue.

**Comment by LeeroyJenks** [comment]
We just came across this issue _sort of_ as some of our devs were passing `html` to `setAppElement`. 
The problem we're now encountering is that the way our site is laid out, passing only one element id will not solve our problem. If we have a layout similar to something below, we'd need to pass an array of ids in order to have expected accessibility behavior.
```html
<body>
  <header id="someheader"></header>
  <main id="somemain"></main>
  <footer id="somefooter"></footer>
</body>
```

**Comment by diasbruno** [comment]
@LeeroyJenks This is available on the latest version.

**Comment by diasbruno** [comment]
Both `appElement` on the component and `setAppElement` can receive a list of elements to hide.

**Comment by jamesonhill** [comment]
Is the official recommendation that `setAppElement` needs to be passed a list of every single focusable element? Something like this?:
```
import ReactModal from 'react-modal';

const root = document.getElementById('root');
  const focusableElements = root.querySelectorAll(
    'a, button, input, select, textarea, [tabindex]:not([tabindex="-1"])'
  );
ReactModal.setAppElement([root, ...focusableElements]);
<div id="root">...a bunch of dynamic content</div>
<div class="PortalWhereModalsAreRendered">...</div>
```

**Comment by diasbruno** [comment]
@jamesonhill The setAppElement tell which elements should be marked as hidden using the aria attribute.

**Comment by jamesonhill** [comment]
@diasbruno yea I get that but it seems like if you add that attribute to the root div and there are elements within that div that are focusable, it fails the WCAG rule "ARIA hidden element must not be focusable or contain focusable elements".

**Comment by diasbruno** [comment]
@jamesonhill There are a few thing we need to look at:

- Modals shouldn't be placed on the element that will receive the attribute
- All focusable elements must be inside of the modal as the UX recommends (tab must cycle through the scoped elements)
- It is possible for the focus to escape the modal, because of browser issues and/or how our `tabbable` functionality handles the scope

**Comment by jamesonhill** [comment]
@diasbruno I have those covered so my issue may be a false positive. Thanks for your insights.

**Comment by diasbruno** [comment]
@jamesonhill Ok. In case you find a different behavoir or a bug, feel freeew to open a new issue. Thanks.
